src/main/java/org/isf/menu/service/MenuIoOperations.java

```diff
@@ -398,6 +386,20 @@ public boolean updateUserGroup(UserGroup userGroup, List<Permission> permissions
 			groupPermissionIoOperationRepository.saveAll(groupPermissions);
 		}
 
-		return updated;
+		return findByCode(userGroup.getCode());
+	}
+
+	public void ensureUserNotDeleted(String username) throws OHServiceException {
+		User entity = repository.findByUserName(username);
+		if (entity == null) {
```

Comment:
The method name says `ensure User Not Deleted`; but the implementation is only checking for a hard delete by checking for `null` (i.e., no longer in the database).    I would have thought that given the method name it would be checking the deleted flag.

Reply:
You're right !